maintenance_plan/models/maintenance.py

```diff
@@ -0,0 +1,219 @@
+                    today_date + period_timedelta)
+
+            plan.next_maintenance_date = next_date
+
+    @api.multi
+    def unlink(self):
+        """ Restrict deletion of maintenance plan should there be maintenance
+            requests of this kind which are not done for its equipment """
+        for plan in self:
+            request = plan.equipment_id.mapped('maintenance_ids').filtered(
+                lambda r: (
+                    r.maintenance_kind_id == plan.maintenance_kind_id and
+                    not r.stage_id.done and
+                    r.maintenance_type == 'preventive')
+            )
+            if request:
+                raise UserError(_('The maintenance plan %s of equipment %s '
+                                  'has generated a request which is not done '
+                                  'yet. You should either set the request as '
+                                  'done, remove its maintenance kind or '
+                                  'delete it first.') % (
+                    plan.maintenance_kind_id.name, plan.equipment_id.name))
+        super(MaintenancePlan, self).unlink()
+
+    _sql_constraints = [
+        ('equipment_kind_uniq', 'unique (equipment_id, maintenance_kind_id)',
+         "You cannot define multiple times the same maintenance kind on an "
+         "equipment maintenance plan!")]
+
+
+class MaintenanceEquipment(models.Model):
+
+    _inherit = 'maintenance.equipment'
+
+    maintenance_plan_ids = fields.One2many(string='Maintenance plan',
+                                           comodel_name='maintenance.plan',
+                                           inverse_name='equipment_id')
+    maintenance_team_required = fields.Boolean(
+        compute='_compute_team_required')
+
+    @api.depends('maintenance_plan_ids')
+    def _compute_team_required(self):
+        for equipment in self:
+            equipment.maintenance_team_required = len(
+                equipment.maintenance_plan_ids) >= 1
+
+    def _create_new_request(self, maintenance_plan):
+        self.ensure_one()
+        self.env['maintenance.request'].create({
+            'name': _('Preventive Maintenance (%s) - %s') %
+            (maintenance_plan.maintenance_kind_id.name,
+             self.name),
+            'request_date': maintenance_plan.next_maintenance_date,
+            'schedule_date': maintenance_plan.next_maintenance_date,
+            'category_id': self.category_id.id,
+            'equipment_id': self.id,
+            'maintenance_type': 'preventive',
+            'owner_user_id': self.owner_user_id.id or self.env.user.id,
+            'technician_user_id': self.technician_user_id.id,
+            'maintenance_team_id': self.maintenance_team_id.id,
+            'maintenance_kind_id': maintenance_plan.maintenance_kind_id.id,
+            'duration': maintenance_plan.duration,
+        })
+
+    @api.model
+    def _cron_generate_requests(self):
```

Comment:
Thanks for your review. 
However, as this method is called by standard ir.cron from maintenance module, I suppose it's better to leave its name as it is, so we don't have to update the existing ir.cron : 
https://github.com/odoo/odoo/blob/10.0/addons/maintenance/data/maintenance_cron.xml#L10